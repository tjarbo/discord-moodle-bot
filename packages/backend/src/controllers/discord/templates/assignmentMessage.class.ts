import { MessageTemplate } from '../message.class';

export interface AssignmentMessageOptions {
  course: string;
  title: string;
  dueDate: string;
}

export class AssignmentMessage extends MessageTemplate {
  readonly template: `**{course} - Neue Abgabe hinzugef√ºgt!**
  :information_source: "{title}" ist abzugeben
  :alarm_clock: bis \`{dueDate}\``;
}