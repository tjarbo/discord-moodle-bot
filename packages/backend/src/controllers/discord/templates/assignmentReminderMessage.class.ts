import { MessageTemplate } from '../message.class';

export interface AssignmentReminderMessageOptions {
  course: string;
  title: string;
}

export class AssignmentReminderMessage extends MessageTemplate {
  readonly template = `**{course} - Erinnerung**:
  :warning: Abgabe von "{title}" **heute** f√§llig!`;
}
