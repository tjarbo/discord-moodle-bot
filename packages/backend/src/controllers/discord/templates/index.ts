export { AssignmentMessage, AssignmentMessageOptions } from './assignmentMessage.class';
export { AssignmentReminderMessage, AssignmentReminderMessageOptions } from './assignmentReminderMessage.class';
export { ResourceMessage, ResourceMessageOptions } from './resourceMessage.class';
export { TokenRequestMessage, TokenRequestMessageOptions } from './tokenRequestMessage.class';
