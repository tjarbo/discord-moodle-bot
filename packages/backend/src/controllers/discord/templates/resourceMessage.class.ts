import { MessageTemplate } from '../message.class';

export interface ResourceMessageOptions {
  course: string;
  title: string;
  link: string;
}

export class ResourceMessage extends MessageTemplate {
  readonly template = `**{course} - Neue Datei**:
  :file_folder: {title}
  :link: {link}`;
}
