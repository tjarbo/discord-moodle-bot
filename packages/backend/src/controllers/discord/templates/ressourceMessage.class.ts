import { MessageTemplate } from '../message.class';

export interface RessourceMessageOptions {
  course: string;
  title: string;
  link: string;
}

export class RessourceMessage extends MessageTemplate {
  readonly template: `**{course} - Neue Datei**:
  :file_folder: {title}
  :link: {link}`;
}