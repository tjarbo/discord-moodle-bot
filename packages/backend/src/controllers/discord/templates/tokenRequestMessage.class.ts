import { MessageTemplate } from '../message.class';

export interface TokenRequestMessageOptions {
  key: string;
}

export class TokenRequestMessage extends MessageTemplate {
  readonly template = `:key: **Es wurde ein Zugangstoken angefordert**
  Zugangstoken lautet: {key}
  Solltest du den Token nicht angefordert haben - Kein Problem, l√∂sche diese Nachricht einfach!`;
}
