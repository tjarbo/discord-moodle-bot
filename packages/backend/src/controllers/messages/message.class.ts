import { compile } from 'handlebars';

/**
 * Abstract class for notification messages
 *
 * @export
 * @abstract
 * @class Message
 */
export abstract class Message {

  /**
   * Set of handlebar.js templates for Markdown messages
   *
   * @protected
   * @abstract
   * @type {Template}
   * @memberof Message
   */
  protected abstract readonly markdownTemplate: string;

  /**
   * Content for
   *
   * ! Content depends on the used template class
   *
   * @protected
   * @abstract
   * @type {*}
   * @memberof Message
   */
  protected abstract context: any;

  /**
   * Return the message as Markdown
   *
   * @readonly
   * @type {string}
   * @memberof Message
   */
  public get Markdown() : string {
    return compile(this.markdownTemplate)(this.context);
  }
}
