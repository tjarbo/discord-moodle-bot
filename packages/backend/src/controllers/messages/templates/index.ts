import { AssignmentMessage } from './assignmentMessage.class';
import { AssignmentReminderMessage } from './assignmentReminderMessage.class';
import { ResourceMessage } from './resourceMessage.class';

export * from './assignmentMessage.class';
export * from './assignmentReminderMessage.class';
export * from './resourceMessage.class';

export type Message = AssignmentMessage | AssignmentReminderMessage | ResourceMessage;
