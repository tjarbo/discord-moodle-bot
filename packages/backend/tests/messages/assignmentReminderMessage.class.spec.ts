import { AssignmentReminderMessage } from '../../src/controllers/messages/templates';

jest.mock('../../src/configuration/environment.ts');

describe('assignmentReminderMessage.class.ts', () => {
  let assignmentMessage: AssignmentReminderMessage;

  beforeAll(() => {
    assignmentMessage = new AssignmentReminderMessage('TEST_COURSE', 'Title');
  });

  it('should match Markdown', () => {
    expect(assignmentMessage.Markdown).toMatchSnapshot();
  });
});
