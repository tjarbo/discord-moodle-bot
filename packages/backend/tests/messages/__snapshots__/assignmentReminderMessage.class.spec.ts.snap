// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignmentReminderMessage.class.ts should match Markdown 1`] = `
"**TEST_COURSE - Reminder**:
⚠️ Assignment \\"Title\\" is **due today**!"
`;
